pkg/expr/nodes.go

```diff
@@ -39,23 +42,15 @@ type baseNode struct {
 }
 
 type rawNode struct {
-	RefID         string `json:"refId"`
-	Query         map[string]interface{}
-	QueryType     string
-	TimeRange     TimeRange
-	DatasourceUID string // Gets populated from Either DatasourceUID or Datasource.UID
+	RefID      string `json:"refId"`
+	Query      map[string]interface{}
+	QueryType  string
+	TimeRange  TimeRange
+	DataSource *models.DataSource
 }
 
 func (rn *rawNode) IsExpressionQuery() bool {
-	if IsDataSource(rn.DatasourceUID) {
-		return true
-	}
-	if v, ok := rn.Query["datasourceId"]; ok {
-		if v == OldDatasourceUID {
-			return true
-		}
-	}
-	return false
+	return IsDataSource(rn.DataSource.Uid)
```

Comment:
IMHO the name `IsDataSource()` is not descriptive/accurate enough. The function checks if a uid points to an expression query. I think it's better to rename it to `IsExpressionQuery()`.

Also, it could be dangerous to access the `Uid` field on a potentially null `rn.DataSource`. When you use the function inside the `*Service.buildGraph()` method you're creating a `rawNode` with a non-nil `DataSource` field. That's ok, but I think we have to take into account that using this method in other places could result in unexpected panics. I would suggest using `expr.IsDataSource(rn.Datasource.Uid)` instead.